integration/tests/basic/optcodes/test_selfdestruct.py

```diff
@@ -0,0 +1,144 @@
+import random
+
+import allure
+import pytest
+
+from integration.tests.basic.helpers.basic import BaseMixin
+
+
+@allure.feature("Optcodes verifications")
+@allure.story("Self-destruction optcode")
+class TestSelfDestructOptcode(BaseMixin):
+
+    @pytest.fixture(scope="function")
+    def destroyable_contract(self):
+        contract, _ = self.web3_client.deploy_and_get_contract(
+            "SelfDestroyable", "0.8.10", self.sender_account)
+        return contract
+
+    @pytest.fixture(scope="function")
+    def contract_caller(self, destroyable_contract):
+        contract, _ = self.web3_client.deploy_and_get_contract(
+            "SelfDestroyable", "0.8.10", self.sender_account,
+            contract_name="SelfDestroyableContractCaller",
+            constructor_args=[destroyable_contract.address])
+        return contract
+
+    def deposit(self, destroyable_contract, sender, amount):
+        tx = self.create_contract_call_tx_object(sender, amount=amount)
+        instruction_tx = destroyable_contract.functions.deposit().buildTransaction(tx)
+        self.web3_client.send_transaction(sender, instruction_tx)
+
+    def destroy(self, destroyable_contract, sender, funds_recipient, amount=None):
+        tx = self.create_contract_call_tx_object(sender, amount=amount)
+        instruction_tx = destroyable_contract.functions.destroy(funds_recipient.address).buildTransaction(tx)
+        receipt = self.web3_client.send_transaction(sender, instruction_tx)
+        assert receipt["status"] == 1
+
+    def check_contract_code_is_empty(self, contract_address):
+        response = self.proxy_api.send_rpc(
+            "eth_getCode",
+            params=[contract_address, "latest"]
+        )
+        assert response["result"] == "0x"
+
+    def test_destroy(self, destroyable_contract):
+        self.deposit(destroyable_contract, self.sender_account, 1)
+        self.deposit(destroyable_contract, self.recipient_account, 1)
```

Comment:
You deposit from account self.recipient_account but never check it balance after destroy